src/snowflake/cli/api/cli_global_context.py

```diff
@@ -218,8 +220,13 @@ def silent(self) -> bool:
         return self._silent
 
     def set_silent(self, value: bool):
+        if self.output_format == OutputFormat.JSON:
+            value = True
         self._silent = value
```

Comment:
I I have feeling this is bit counter intuitive. You set something but you don't. I would personally move this into getter of `self._silent`, WDYT?

Reply:
Done